wallettool/src/main/java/org/bitcoinj/wallettool/WalletTool.java

```diff
@@ -325,53 +321,78 @@ public static void main(String[] args) {
 
     @Override
     public Integer call() throws IOException, BlockStoreException {
-        ActionEnum action;
-        try {
-            action = ActionEnum.valueOf(actionStr.toUpperCase().replace("-", "_"));
-        } catch (IllegalArgumentException e) {
-            System.err.println("Could not understand action name " + actionStr);
-            return 1;
-        }
+        ActionEnum action = validateAction(actionStr);
+        if (action == null) return 1;
 
-        if (debugLog) {
-            BriefLogFormatter.init();
-            log.info("Starting up ...");
-        } else {
-            // Disable logspam unless there is a flag.
-            java.util.logging.Logger logger = LogManager.getLogManager().getLogger("");
-            logger.setLevel(Level.SEVERE);
-        }
+        setupLogging();
```

Comment:
You should pass `debugLog` as a parameter to his method, rather than have it reference a member. This will begins the separation of pico-cli UX from the actual operation of the wallet implementation.